### DISCUSSION TITLE
Save scene

### DISCUSSION THREAD

**Discussion by SinicinDenis** [pokepetter/ursina#633] [Q&A]
Hi, how do I save a scene for later loading?

**Comment by OfficialAhmed** [comment]
I dont think there is a way to do it in Ursina I have searched the whole documentation couldn't find an easy way other than using basic I/O to a JSON file save the player info i.e. state, position etc.

Otherwise, my approach would be through `OOP` to save the entire class to `JSON` and loading it later so you dont have to init everything from the start here's a snippet 

```python
class Player:
    def __init__():
        # init player info
        self.player_pos = Vect3(0,0,0)
        self.score = 0

    def to_dict(self):
        # return a dict of all player state
        return {'player_pos': self.player_pos, 'score': self.score}

    def save(self, file_path):
        with open(file_path, 'w+') as file:
            json.dump(self.to_dict(), file)

    @classmethod
    def load(cls, file_path):
        with open(file_path, 'r') as file:
            data = json.load(file)
        return cls(**data)       # Reconstruct the class instance using the loaded data
```

to Load

```python
loaded_class = Player.load('savegame.json')
```

Using this approach you can save as many classes as you want to a file

**Reply by SinicinDenis** [reply]
yes, that's how I implemented it, the current is through a txt file. .transform - pos rot scale, you can also call .attributes, it will return the keys to the dictionary
